lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -300,7 +300,9 @@ def run_async(command)
       raise 'Async command already in progress' if @process_manager&.started?
 
       @user_method.reset
-      initialize_command(*get_args(command, true))
+      full_cmd = get_args(command, true)
+      @logger.debug("running command: #{full_cmd.join(' ')}")
```

Comment:
The first element of `full_cmd` might contain a hash where there might be effective user password under `SSHPASS` key. That should never end up in the logs, no matter the level

Reply:
Absolutely agree.  This PR started as a local hack I've made debugging several times over the last few years and wasn't deeply though through.

I realize that ultimately the interface of `Process#spawn` has to be satisfied but it is a bit unsettling to have secrets mixed into the same array as the command arguments.  It feels like there should be a standard-ish secret or confidential string type that returns something like "<SECRET>" or raises an exception on `#to_s` but a brief search of rubygems hasn't found such a thing... is there such a thing?

One option would be to filter the args array and remove any hashes as part of the interpolation in the debug statement.  How would you feel about splitting the env vars and command line args up into independent data structures that are either merged before being passed to `Proxy::Dynflow::ProcessManager#initialize` or changing the interface of that constructor?

Reply:
> is there such a thing?

I'm not aware of any standard solution for this, although a custom one should be fairly straightforward to implement.

> How would you feel about splitting the env vars and command line args up into independent data structures that are either merged before being passed

I would be fine with this

> or changing the interface of that constructor?

I'd like to avoid this. I'd say coordinating changes landing in three different projects and breaking the standard interface ala spawn just to make something, that can be done in a single project with minimal amount of effort, slightly easier is not worth it.